R/extract.R

```diff
@@ -73,7 +75,13 @@ get_chunk_positions <- function(pattern, lines) {
   # only keep those blocks that contain at least one line of code
   keep <- which(ends - starts > 1L)
 
-  list(starts = starts[keep], ends = ends[keep])
+  starts <- starts[keep]
+  ends <- ends[keep]
+
+  # This is _somewhat_ fragile... better may be to do gsub("```.*", "", pattern$chunk.begin),
+  #   but that somehow feels roughly as fragile (and less readable).
+  start_indent_matches <- gregexpr("^[\t >]*", lines[starts], perl = TRUE)
```

Comment:
Are there other indents apart from spaces and blockquotes?

Reply:
this is extracted from the knitr pattern, so knitr won't recognize anything else

Reply:
OK, wrote it to be explicitly based on the knitr pattern to make that a bit clearer.